### PR TITLE
Make simulators Dockerfiles arch agnostic

### PR DESCRIPTION
**Pull request by mynameisdaniil**
I've got stuck trying to run simulators. @kdeme quickly helped me, but I think we can do better than commenting/uncommetings lines.
`aarch64->arm64` needed because some systems report `aarch64` and some `arm64` while ubuntu only uses `arm64` as an architecture name.

### CODE REVIEW

**Comment by mynameisdaniil** [comment]
@kdeme, @ogenev please take a look

File: simulators/portal-interop/Dockerfile [hunk 1]
```diff
@@ -18,12 +18,9 @@ FROM ubuntu:22.04
 
 RUN apt update && apt install wget -y
 
-# Use these for amd-based devices
-RUN wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.0g-2ubuntu4_amd64.deb
-RUN dpkg -i libssl1.1_1.1.0g-2ubuntu4_amd64.deb
-# Use these for arm-based devices
-#RUN wget http://ports.ubuntu.com/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2_arm64.deb
-#RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2_arm64.deb
+RUN export ARCH=$(uname -m | sed "s/aarch64/arm64/") \
```
Comment (kdeme) [inline simulators/portal-interop/Dockerfile:21]: Did you test this on an x86_64 machine?

Because the resulting string does not work for me.  (uname = `x86_64`).

**Review by kdeme — CHANGES_REQUESTED** [review]

File: simulators/rpc-compat/Dockerfile [hunk 2]
```diff
@@ -18,12 +18,14 @@ FROM ubuntu:22.04
...
-#RUN wget http://ports.ubuntu.com/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2_arm64.deb
-#RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2_arm64.deb
+RUN export ARCH=$(uname -m | sed -e "s:aarch64:arm64:" -e "s:x86_64:amd64:") \
+  && export BASE=$(case $ARCH in ("amd64") echo "nz2.archive" ;; ("arm64") echo "ports" ;; (*) echo "ports" ;; esac) \
+  && wget http://${BASE}.ubuntu.com/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2_${ARCH}.deb \
+  && dpkg -i libssl1.1_1.1.1f-1ubuntu2_${ARCH}.deb
+
+RUN 
+
+RUN sudo apt-get install
```
Comment (KolbyML) [inline simulators/rpc-compat/Dockerfile:28]:
```suggestion
```

We should probably remove these since they don't do anything?

Comment (KolbyML) [inline simulators/rpc-compat/Dockerfile:28]:
```suggestion
```

We should probably remove these since they don't do anything?

Comment (mynameisdaniil) [reply simulators/rpc-compat/Dockerfile]: Sorry, I was updating my branch to test on another laptop.

File: simulators/portal-interop/Dockerfile (hunk 1 above)
Comment (mynameisdaniil) [reply simulators/portal-interop/Dockerfile]: I should probably stop commiting at 5 in a morning ;-)

**Comment by mynameisdaniil** [comment]
@kdeme @KolbyML Please take a look again.
So, here's a story. We shouldn't just download and install old packages for a number of reasons. Main of which is that we already have a package manager that solves all the problems. Unfortunately `libssl1.1` is no longer available in mainstream linux distros. Best solution would be to update dependencies and use `libssl3`, unfortunately `reqwest` (one of the dependencies) still does not work with `libssl3`. Instead of re-writing or replacing it (from what I can see `libssl3` and Rust does not mix well [[#1]](https://www.reddit.com/r/rust/comments/ukhjzk/misadventures_with_ubuntu_2204_and_libssl3/)) another solution would be to switch container OS to something that supports `libssl1.1`. Obvious choice is `alpine`, unfortunately it crashes in this environment, and I'm not the only one with this specific problem ([[#2]](https://github.com/seanmonstar/reqwest/issues/1528)). Another option is to switch to  debian (since it is default rust container OS anyway). Latest release that supports `libssl1.1` is `bullseye`, fortunately Rust provides build containers with `bullseye`.

TL;DR: We switch to debian bullseye instead of ubuntu 22:04 since it uses `libssl1.1` by default. Better solution would be to replace `reqwest` with something else or make PR into `reqwest` to support `libssl3`.

**Comment by kdeme** [comment]
Sounds good to me. Definitely prefer this over these custom ubuntu port packages :+1: 

@ogenev @KolbyML wdyt?

**Review by kdeme — APPROVED** [review]

**Comment by KolbyML** [comment]
libssl1.1 is already past its end of life https://www.openssl.org/blog/blog/2023/06/15/1.1.1-EOL-Reminder/

If reqwest relies on libssl1.1 it kind of just makes me want to change which httpclient we use.

Other then that, the PR looks fine

**Comment by KolbyML** [comment]
Here is an alternative solution https://github.com/ethereum/portal-hive/pull/103

This solution
- Doesn't use EOL libssl version
- Decreases compiled dependencies, decreasing compile time
- Still uses reqwest, just disables TLS since we don't need it as we only make http requests.
- Same result in code being platform agnostic

**Comment by mynameisdaniil** [comment]
@KolbyML yep, that is even better
